processing/src/main/java/org/apache/druid/segment/data/FixedIndexedWriter.java

```diff
@@ -46,14 +46,16 @@ public class FixedIndexedWriter<T> implements DictionaryWriter<T>
   private final Comparator<T> comparator;
   private final ByteBuffer scratch;
   private final ByteBuffer readBuffer;
-  private int numWritten;
+  private final boolean isSorted;
+  private final int width;
+
+  private int cardinality = 0;
```

Comment:
nit: `FixedIndexed` doesn't really have to be sorted, this only seems like a cardinality if it is sorted (and the other `Indexed` implementation writers still using `numWritten`)

Reply:
Yeah, maybe we didn't need the rename to `cardinality`.  At the same time, whether the variable is named `cardinality` or `numWritten`, it's still returned from the `getCardinality()` method...

Reply:
ah yea, this is just a nitpick, i was thinking it would be kind of nice if all of the `Indexed` writer implementations are consistent. `Indexed` itself however doesn't actually have a `getCardinality()`, it has a `size()` :D.

 it is only the `DictionaryWriter` interface that calls it cardinality which totally makes sense there, but the writers aren't always strictly writing dictionaries either, so :shrug: I don't feel super strongly either way other than consistent naming between them so its easier to look at implementations and see the differences while having a common language